alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -471,8 +472,27 @@ void RunningPlan::deactivate()
     } else {
         _ae->getTeamObserver().notifyRobotLeftPlan(_activeTriple.abstractPlan);
     }
+
     revokeAllConstraints();
     deactivateChildren();
+
+    /*
+     * Create instance of scheduler, later use shared instance for actual scheduling.
+     */
+    scheduler::Scheduler scheduler;
```

Comment:
yes this should be part of the engine and all running plans should get it from there.